File: Algorithm by Swift/Algorithm by Swift/Array/2577.swift
//
//  main.swift
//  Algorithm by Swift
//
//  Created by zaehorang on 2024/02/13.
//  https://www.acmicpc.net/problem/2577

import Foundation

let a = Int(readLine()!)!
let b = Int(readLine()!)!
let c = Int(readLine()!)!

let d = "\(a * b * c)"

var occur = [Int](repeating: 0, count: 10)

for i in d {
    let index = String(i)
    
    occur[Int(index)!] += 1
    
}

for i in occur {
    print(i)
}

// 위와 동일

/*
 var abc = [Int]()
 var arr = [Int](repeating: 0, count: 10)
 for _ in 0..<3 {
     abc.append(Int(readLine()!)!)
 }
 
 String(abc.reduce(1, *)).forEach {
     arr[Int(String($0))!] += 1
 }

 arr.forEach {
     print($0)
 }
 
 */

File: Algoritm by CPP/Algoritm by CPP/Array/2577.cpp
//
//  main.cpp
//  Algoritm by CPP
//
//  Created by zaehorang on 2024/02/13.
//  https://www.acmicpc.net/problem/2577

#include <bits/stdc++.h>
using namespace std;

int main(void){
    ios::sync_with_stdio(0);
    cin.tie(0);
    
    int a, b, c;
    int d[10]={};
    
    cin >> a >> b >> c;
    
    int t=a*b*c;
    
    // swift는 문자열로 순회를 돌 수 있지만 cpp는 불가능
    while (t>0) {
        d[t%10]++;
        t/=10;
    }
    
    for (int i=0; i<10; i++) cout << d[i] << '\n';
    
}
